src/common/TreeOps.h

```diff
@@ -57,6 +57,7 @@ class TermVisitor {
             PTRef currentRef = currentEntry.term;
             if (not cfg.previsit(currentRef)) {
                 toProcess.pop_back();
+                done[Idx(logic.getPterm(currentRef).getId())] = 1;
```

Comment:
This is needed to prevent duplicate calls to `previsit` if it returned `false` the first time.
I think this makes sense regardless of the other changes.